mailbox/memory/src/test/java/org/apache/james/mailbox/inmemory/manager/InMemoryIntegrationResources.java

```diff
@@ -156,12 +168,16 @@ public InMemoryIntegrationResources create() {
                 storeRightManager,
                 quotaComponents,
                 index,
-                createHooks(sessionProvider, mailboxSessionMapperFactory));
+                createHooks(preInstanciationStage));
 
             eventBus.register(listeningCurrentQuotaUpdater);
             eventBus.register(new MailboxAnnotationListener(mailboxSessionMapperFactory, sessionProvider));
 
-            return new InMemoryIntegrationResources(manager, storeRightManager, new InMemoryMessageId.Factory(), currentQuotaManager, quotaRootResolver, maxQuotaManager, quotaManager);
+            if (index instanceof MailboxListener.GroupMailboxListener) {
```

Comment:
why?

Reply:
This was done in order to rely on IntegrationResources for Search index tests.

I did not come up with better design so I can very well drop this part of the refactoring.

Technical explanation:

You should not loose the type of the search index.

Maybe it should be a caller responsibility Akka:

```
InMemoryIntegrationResources resources = new InMemoryIntegrationResources.Factory()
            .withSearchIndex(preInstanciationStage -> new ElasticSearchListeningMessageSearchIndex(/* ... */)
            .create();

resources.getMailboxManager().getEventBus().register((MailboxListener.GroupMailboxListener) resources.getMailboxManager().getMessageSearchIndex());
```

WDYT ?

Reply:
I found a better way by exposing the knowledge of  `ListeningSearchIndex` through the API of `InMemoryIntegrationResources`